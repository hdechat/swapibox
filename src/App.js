import React, { Component } from 'react';
import './App.css';
import cleanFilmData from './cleaners.js'
import filmData from './mockFilmData.js'

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      textCrawl: ''
    } 
  }

  componentDidMount() {
    const textCrawl = cleanFilmData(filmData);
    this.setState({ textCrawl });
  }

  render() {
    return (
      <div>
      </div>
    );
  }
}

export default App;
