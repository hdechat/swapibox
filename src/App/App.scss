aside {
  background-color: black;
  color: white;
  height: 850px;
  opacity: .9;
  overflow: hidden;
  position: absolute;
  width: 100%;
  z-index: 10;
}

main {
  text-align: center;
}

.background {
  background: url('../assets/background-sw.png');
  background-size: cover;
  font-family: 'Pathway Gothic One', sans-serif;
  height: 850px;
}

.buttons {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  padding-top: 15rem;
}

.loading-gif {
  width: 85%;
}