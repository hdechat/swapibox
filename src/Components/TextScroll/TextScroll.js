import React from 'react'
import PropTypes from 'prop-types'
import './TextScroll.css'

const TextScroll = ({ textCrawl }) => {
  return (
    <div>
      <div className="textScroll">
        <p>{textCrawl.text}</p>
        <p>{textCrawl.title}</p>
        <p>{textCrawl.released}</p>
      </div>
      <button onClick={() => document.querySelector('aside').style.display = 'none'}>Click</button>
    </div>
  )
}

TextScroll.propTypes = {
  textCrawl: PropTypes.object.isRequired
}

export default TextScroll
