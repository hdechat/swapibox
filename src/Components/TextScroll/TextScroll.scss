.textcrawl-container {
  background: url("../../assets/starry-night.png");
  height: 100%;
  overflow: hidden;
  width: 100%;
}

.star-wars {
  border: none;
  color: #feda4a;
  display: flex;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 500%;
  font-weight: 600;
  height: 800px;
  justify-content: center;
  letter-spacing: 6px;
  line-height: 150%;
  perspective: 400px;
  position: relative;
  text-align: justify;
  width: 100%;
}

.crawl {
  position: relative;
  transform-origin: 50% 10%;
  animation: crawl 70s linear;
}



@keyframes crawl {
  0% {
    top: 10em;
    transform: rotateX(40deg) translateZ(0);
  }
  85% { 
    top: -6000px;
    transform: rotateX(0deg) translateZ(-2500px);
  }
}
