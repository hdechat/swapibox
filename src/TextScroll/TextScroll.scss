.textScroll{
  position: absolute;
  left: 30%;
  width: 40%;
  animation-name: crawl;
  animation-duration: 30s;
}

@keyframes crawl {
    0%   { bottom:0%; transform:perspective(0px) rotateX(0deg)}
    100% {bottom:80%; transform:perspective(0px) rotateX(0deg)}
}

