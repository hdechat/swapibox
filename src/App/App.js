import React, { Component } from 'react';
import './App.css';
import TextScroll from '../TextScroll/TextScroll.js';
import Button from '../Button/Button.js';
import CardContainer from '../CardContainer/CardContainer.js';

export default class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      textCrawl: {},
      cardList: [],
      favorites: [],
      //Refactor: inital fetch to get links to categories
      // peopleLink: null,
      // vehicleLink: null,
      // planetsLink: null
    } 
  }

  async componentDidMount() {
    const data = await this.fetchTextCrawl();
    this.setState(
      { textCrawl: {
        'title': data.title,
        'text': data.opening_crawl,
        'released': data.release_date
      }}
    );
  }

  fetchTextCrawl = async () => {
    let randomFilm = Math.floor(Math.random() * 7 + 1);
    const response = await fetch(`https://swapi.co/api/films/${randomFilm}`);
    return await response.json();
  }

  clickedCard = (faved, card) => {
    const upDatedCard = {...card, favorited: faved}
    const upDatedCardList = this.state.cardList
      .map(person => card.name === person.name ? upDatedCard : person);
    this.setState({ cardList: upDatedCardList });

    if(faved) {
      this.setState({ favorites: [...this.state.favorites, upDatedCard] });
    } else {
      const upDatedFaves = this.state.favorites.filter(fave => fave.name !== card.name);
      this.setState({ favorites: upDatedFaves});
      document.querySelector('.favorites').classList.value.includes('active') ? this.setState({ cardList: upDatedFaves }) : '';
    }
  }

  fetchHomeData = async ({ homeworld, name }) => {
    const response = await fetch(homeworld);
    const data = await response.json();
    return await {name: name, homeworld: data.name, population: data.population};
  }

  fetchSpeciesData = async (person) => {
    const response = await fetch(...person.species);
    const data = await response.json();
    return await {species: data.name};
  }

  createPeopleData = (people) => {
    const peopleData = people.reduce((arr, person) => {
      if(person.species.length) {
        const species = this.fetchSpeciesData(person)
      return [...arr, species, this.fetchHomeData(person)];
      } else {
        return [...arr, {species: 'unknown'}, this.fetchHomeData(person)];
      }
    }, []);
    return Promise.all(peopleData);
  }

  fetchPeopleList = async () => {
    let peopleCards = [];

    //there are 9 pages of data available. to fetch all 9 pages change the iteration length to 10.
    for(let i = 1; i < 2; i++) { 
      const response = await fetch(`https://swapi.co/api/people/?page=${i}`);
      const data = await response.json();
      const cards = await this.createPeopleData(data.results);

      while(cards.length) {
          peopleCards.push({...cards.shift(), ...cards.shift(), favorited: false});
      }
    }

    const mergedCards = peopleCards.map(card => {
      const favorited = this.state.favorites.find(fave => fave.name === card.name);
      return (favorited ? favorited : card);
    });

    this.setState({ cardList: mergedCards });
  }

  showFavorites = () => {
    // if(!this.state.favorites.length) {
    //   document.querySelector('section').innerHTML = '<h1>No Favorite Cards</h1>';
    // }
    this.setState({ cardList: this.state.favorites });
  }
  
  render() {
    return (
      <div className="background">
        <aside>
          <TextScroll textCrawl={this.state.textCrawl} />
        </aside>
        <main>
          <div className="buttons">
            <Button category={"favorites"} callback={this.showFavorites} />
            <Button category={"People"} callback={this.fetchPeopleList} />
            <Button category={"Vehicles"} callback={()=>{}} />
            <Button category={"Planets"} callback={()=>{}} />
          </div>
          <CardContainer clickedCard={this.clickedCard} cardList={this.state.cardList}/>
        </main>
      </div>
    );
  }
}