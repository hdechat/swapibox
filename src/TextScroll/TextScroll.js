import React from 'react';
import PropTypes from 'prop-types';
import './TextScroll.scss';

const TextScroll = ({ textCrawl }) => {
  return (
    <div>
      <p>{textCrawl.text}</p>
      <p>{textCrawl.title}</p>
      <p>{textCrawl.released}</p>
    </div>
  )
}

TextScroll.propTypes = {
  textCrawl: PropTypes.object
}

export default TextScroll;
