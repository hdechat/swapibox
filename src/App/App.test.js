import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import App from './App.js';
import TextScroll from '../TextScroll/TextScroll.js'

describe('App', () => {
  let app;

  beforeEach(() => {
    app = shallow(<App />);
  })

  it('matches snapshot', () => {
    expect(app).toMatchSnapshot();
  });

  xit('sets state.textCrawl on initial load', () => {
    expect(app.state('textCrawl')).toBeDefined();
  });

  it('should add card to favoriteList when parameter is true', () => {
    const fave = true;
    const card = {name: "Name"};

    app.instance().clickedCard(fave, card);

    expect(app.favorites).length.toEqual(1);
  })

  xit('should delete card from favoriteList when parameter is false', () => {
  });
});