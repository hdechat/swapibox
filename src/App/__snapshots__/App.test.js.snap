// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <aside>
          <TextScroll
                    textCrawl={Object {}}
          />
</aside>,
        <main>
          <div
                    className="buttons"
          >
                    <Button
                              callback={[Function]}
                              category="favorites"
                    />
                    <Button
                              callback={[Function]}
                              category="People"
                    />
                    <Button
                              callback={[Function]}
                              category="Vehicles"
                    />
                    <Button
                              callback={[Function]}
                              category="Planets"
                    />
          </div>
          <CardContainer
                    cardList={Array []}
          />
</main>,
      ],
      "className": "background",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <TextScroll
            textCrawl={Object {}}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "textCrawl": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "aside",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="buttons"
>
              <Button
                            callback={[Function]}
                            category="favorites"
              />
              <Button
                            callback={[Function]}
                            category="People"
              />
              <Button
                            callback={[Function]}
                            category="Vehicles"
              />
              <Button
                            callback={[Function]}
                            category="Planets"
              />
</div>,
            <CardContainer
              cardList={Array []}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Button
                  callback={[Function]}
                  category="favorites"
/>,
                <Button
                  callback={[Function]}
                  category="People"
/>,
                <Button
                  callback={[Function]}
                  category="Vehicles"
/>,
                <Button
                  callback={[Function]}
                  category="Planets"
/>,
              ],
              "className": "buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "callback": [Function],
                  "category": "favorites",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "callback": [Function],
                  "category": "People",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "callback": [Function],
                  "category": "Vehicles",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "callback": [Function],
                  "category": "Planets",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "cardList": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <aside>
            <TextScroll
                        textCrawl={Object {}}
            />
</aside>,
          <main>
            <div
                        className="buttons"
            >
                        <Button
                                    callback={[Function]}
                                    category="favorites"
                        />
                        <Button
                                    callback={[Function]}
                                    category="People"
                        />
                        <Button
                                    callback={[Function]}
                                    category="Vehicles"
                        />
                        <Button
                                    callback={[Function]}
                                    category="Planets"
                        />
            </div>
            <CardContainer
                        cardList={Array []}
            />
</main>,
        ],
        "className": "background",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <TextScroll
              textCrawl={Object {}}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "textCrawl": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "aside",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="buttons"
>
                <Button
                                callback={[Function]}
                                category="favorites"
                />
                <Button
                                callback={[Function]}
                                category="People"
                />
                <Button
                                callback={[Function]}
                                category="Vehicles"
                />
                <Button
                                callback={[Function]}
                                category="Planets"
                />
</div>,
              <CardContainer
                cardList={Array []}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Button
                    callback={[Function]}
                    category="favorites"
/>,
                  <Button
                    callback={[Function]}
                    category="People"
/>,
                  <Button
                    callback={[Function]}
                    category="Vehicles"
/>,
                  <Button
                    callback={[Function]}
                    category="Planets"
/>,
                ],
                "className": "buttons",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "callback": [Function],
                    "category": "favorites",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "callback": [Function],
                    "category": "People",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "callback": [Function],
                    "category": "Vehicles",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "callback": [Function],
                    "category": "Planets",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "cardList": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
